src/orquestra/vqa/estimation/cvar.py

```diff
@@ -62,14 +63,14 @@ def __call__(
         )
 
         if self.use_exact_expectation_values:
-            if not isinstance(backend, QuantumSimulator):
+            if not isinstance(runner, BaseWavefunctionSimulator):
                 raise TypeError(
                     "In order to use exact expectation values "
                     "you need to use QuantumSimulator."
                 )
 
             wavefunctions_list = [
-                backend.get_wavefunction(circuit) for circuit in circuits
+                runner.get_wavefunction(circuit) for circuit in circuits
```

Comment:
Since now we are using protocols, we shouldn't perform an `isinstance` check. Instead, I think we should try obtaining wavefunctions lists, catch `AtributeError`, and reraise `TypeError` with our custom message. Something along the lines of (not tested, writing it in a browser):

```python
if self.use_exact_expectation_values:
    try:
        wavefunctions_list = [
            runner.get_wavefunction(circuit) for circuit in circuits
        ]
    except AttributeError:  # this means there was not `get_wavefunction` attribute
        raise TypeError(
            "In order to use exact expectation values "
            "you need to use QuantumSimulator."
         )
````